toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -424,16 +424,18 @@ func WaitForDevicesToSettle() error {
 
 // CreatePartitions creates partitions on the specified disk according to the disk config
 func CreatePartitions(diskDevPath string, disk configuration.Disk, rootEncryption configuration.RootEncryption,
-	readOnlyRootConfig configuration.ReadOnlyVerityRoot,
+	readOnlyRootConfig configuration.ReadOnlyVerityRoot, diskKnownToBeEmpty bool,
 ) (partDevPathMap map[string]string, partIDToFsTypeMap map[string]string, encryptedRoot EncryptedRootDevice, readOnlyRoot VerityDevice, err error) {
 	const timeoutInSeconds = "5"
 	partDevPathMap = make(map[string]string)
 	partIDToFsTypeMap = make(map[string]string)
 
 	// Clear any old partition table info to prevent errors during partition creation
-	_, stderr, err := shell.Execute("sfdisk", "--delete", diskDevPath)
-	if err != nil {
-		logger.Log.Warnf("Failed to clear partition table. Expected if the disk is blank: %v", stderr)
+	if !diskKnownToBeEmpty {
+		_, stderr, err := shell.Execute("sfdisk", "--delete", diskDevPath)
+		if err != nil {
+			logger.Log.Warnf("Failed to clear partition table. Expected if the disk is blank: %v", stderr)
```

Comment:
Are there cases where this can still happen with the given error message `Expected if disk is blank`?

Reply:
Yes. If you are dealing with a real disk, then the disk might still have data on it from the previous install. Or the disk might be completely clean (e.g. new).

Reply:
I think I didn't get this part. What does `diskKnownToBeEmpty` mean exactly if it's still possible for the disk to be blank (according to the warning message) even when `diskKnownToBeEmpty` is `false`? And since at this point we're interesting in making sure the disk is cleared regardless, if it had any data on it or not, would it be possible to check that first and then expect no errors from `sfdisk` instead of passing another argument and having to still potentially ignore legitimate errors.

Reply:
If `diskKnownToBeEmpty` is `true`, then the disk is guaranteed to be empty. But if `diskKnownToBeEmpty` is `false`, then the status of the disk is not known.

I don't think there is a reliable way to tell if an existing disk has data or not. It's just a blob of bits at the end of the day.

Reply:
My goal for this change is to reduce number of the situations where the "Failed to clear partition table. Expected if the disk is blank" is seen.